src/utilities/util_dump.c

```diff
@@ -805,15 +842,328 @@ dump_all_records(WT_CURSOR *cursor, bool reverse, bool json)
+    WT_DECL_RET;
+    WT_SESSION *session;
+    uint64_t bookmark_index, window;
+    int i, exact, num_args;
+    char *args[MAX_ARGS], *bookmarks[MAX_BOOKMARKS];
+    char *first_arg, user_input[MAX_ARGS], *current_arg;
+    const char *key, *value;
+    bool once, search_near;
+
+    session = cursor->session;
+    once = search_near = false;
+    i = exact = num_args = 0;
+    bookmark_index = window = 0;
+    for (i = 0; i < MAX_BOOKMARKS; ++i)
+        bookmarks[i] = NULL;
+
+    printf("**************************\n");
+    printf("Explore mode for %s.\n", uri);
+    printf("**************************\n");
+    printf("Enter 'h' for help, 'q' to exit.\n\n");
+
+    while (ret == 0) {
+        i = num_args = 0;
+        fgets(user_input, MAX_ARGS, stdin);
+
+        /* Remove new line character. */
+        user_input[strlen(user_input) - 1] = '\0';
+        if (strlen(user_input) == 0 && !once)
+            continue;
+        once = true;
+
+        /* Time to exit. */
+        if (strlen(user_input) == 1 && user_input[0] == 'q')
+            break;
+
+        /* Parse the input. */
+        current_arg = strtok(user_input, " ");
+        while (current_arg != NULL) {
+            args[i] = malloc(ARG_BUF_SIZE);
+            strcpy(args[i++], current_arg);
+            ++num_args;
+            current_arg = strtok(NULL, " ");
+        }
+
+        first_arg = args[0];
+        switch (first_arg[0]) {
+        /* Cursor info. */
+        case 'a':
+            ret = cursor->get_key(cursor, &key);
+            if (ret != 0 && ret != EINVAL)
+                return (util_cerr(cursor, "get_key", ret));
+            else if (ret == EINVAL) {
+                printf("Error: the cursor needs to be positioned.\n");
+                ret = 0;
+                break;
+            }
+
+            if ((ret = cursor->get_value(cursor, &value)) != 0)
+                return (util_cerr(cursor, "get_value", ret));
+
+            printf("Current position:\n");
+            printf("key:%s\n", key);
+            printf("value:%s\n", value);
+            break;
+        /* Bookmarks. */
+        case 'b':
+            if (strcmp(first_arg, "b") == 0) {
+                /* List existing bookmarks. */
+                if (num_args < 2) {
+                    printf("List of bookmarks:\n");
+                    for (i = 0; i < MAX_BOOKMARKS; ++i) {
+                        if (bookmarks[i] != NULL)
+                            printf("#%d: %s\n", i, bookmarks[i]);
+                    }
+                }
+                /* Jump to the bookmark. */
+                else {
+                    if (util_str2num(session, args[1], true, &bookmark_index) != 0 ||
+                      bookmark_index >= MAX_BOOKMARKS) {
+                        printf("Error: please indicate a value between 0 and %d\n", MAX_BOOKMARKS);
+                        break;
+                    }
+                    key = bookmarks[bookmark_index];
+                    if (key == NULL) {
```

Comment:
There is nothing wrong here, but in the code I've read in the code base to date the following appears to be preferred:
```suggestion
                    if ((key = bookmarks[bookmark_index]) == NULL) {
```

Reply:
Updated in 3a8666c193a870d2a071dd349105c98635176418